homeassistant/components/ecovacs/sensor.py

```diff
@@ -237,6 +274,32 @@ async def on_event(event: LifeSpanEvent) -> None:
         self._subscribe(self._capability.event, on_event)
 
 
+class EcovacsPositionSensor(
+    EcovacsDescriptionEntity[Capabilities, CapabilityEvent[PositionsEvent]],
+    SensorEntity,
+):
+    """Position sensor."""
+
+    entity_description: EcovacsPositionSensorEntityDescription
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        await super().async_added_to_hass()
+
+        async def on_event(event: PositionsEvent) -> None:
+            for position in event.positions:
+                if position.type == self.entity_description.position_type:
+                    coordinates = {
+                        ATTRIBUTE_POSITION_X: position.x,
+                        ATTRIBUTE_POSITION_Y: position.y,
+                    }
+                    self._attr_native_value = json.dumps(coordinates)
```

Comment:
i'm not sure if a json string should be used for a user visible sensor state in the UI 🤔

Reply:
I can replace the JSON with a comma separated pair of values (like x,y).

Reply:
A sensor state should be one value, which can directly be used and shown in the UI.
In your case the state is combined with multiple values, which is not allowed

Reply:
What do you suggest for this specific case? A sensor with datetime as value (last update) and coordinates as attribuites can be a possible solution?

Reply:
This exactly describes an event entity 😉